util-core/src/main/scala/com/twitter/util/Batcher.scala

```diff
@@ -0,0 +1,24 @@
+package com.twitter.util
+
+/** Provides a clean, lightweight interface for working with a batched [[com.twitter.util.Future]] */
+class Batcher[In, Out] private[util](
+  executor: BatchExecutor[In, Out]
+)(
+  implicit timer: Timer
+) extends Function1[In, Future[Out]] { batcher =>
```

Comment:
I don't have a strong feeling about this.  I think the reason I went with Function1 was that I was having trouble getting it to compile the other way.  I've since found that I can get it to work only with parens:

```scala
) extends (In => Future[Out]) { batcher =>
```

@mosesn @vkostyukov please let me know if you want me to make this change